src/messaging/messages/message.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @module Message
+ */
+import uuidv4 from 'uuid/v4'
+
+/** A base class for all types of messages */
+export default class Message {
+  /**
+   * @param {object} body - A plain JS object (with no methods) representing a body of the message.
+   */
+  constructor (body) {
+    /**
+     * A message's role (@see {@link Message.roles}). Will be defined in descendants.
+     *
+     * @type {string | undefined}
+     */
+    this.role = undefined
+
+    /**
+     * A type of the message. Used to distinguish one kind of message from the other (@see {@link Message.types}).
+     *
+     * @type {string | undefined}
+     */
+    this.type = Message.types.GENERIC
+
+    /**
+     * A unique identifier of the message.
+     *
+     * @type {string}
+     */
+    this.ID = uuidv4()
+
+    /**
+     * An object with no methods representing a message body.
+     *
+     * @type {object}
+     */
+    this.body = body
+  }
+}
+
+/**
+ * Specifies whether a message is request or response.
+ */
+Message.roles = {
+  REQUEST: 'Request',
+  RESPONSE: 'Response'
+}
+
+/**
+ * Specifies a message type: what kind of message it is and what purpose it serves.
+ */
+Message.types = {
+  GENERIC: 'Generic'
```

Comment:
Probably we won't have many message types, but it's good to have for extensibility. Overall, I think it's simpler to have one type of message that will deliver different content (content analysis will be done to determine what actions are requested) than multiple types of messages. From my point of view multiple message types are less flexible and harder to maintain. What do you think?

Reply:
yes I think we should avoid unnecessary complexity until there is a demonstrated need for it.